### PR TITLE
CM-230: tasks refactor

### PR DESCRIPTION
**Pull request by jdolkowski**
https://openimis.atlassian.net/browse/CM-230

This PR will improve modularity of the module and adds beneficiary crud searcher.

It needs:
https://github.com/openimis/openimis-be-tasks_management_py/pull/16
https://github.com/openimis/openimis-be-social_protection_py/pull/32
https://github.com/openimis/openimis-fe-social_protection_js/pull/47

### CODE REVIEW

File: src/components/TaskPreviewTable.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  },
+});
+
+function TaskPreviewTable({
+  classes,
+  previewItem,
+  itemFormatters,
+  tableHeaders,
+}) {
+  const { fetchingTasks, errorTasks } = useSelector((state) => state?.socialProtection);
```
Comment (sniedzielski) [inline src/components/TaskPreviewTable.js]: Why we have here reference to 'socialProtection' after your refactor? Is this one on purpose? Task preview should be more 'general' I think. Let me know if this is possible to fix this at this stage

Comment (sniedzielski) [reply src/components/TaskPreviewTable.js]: Should be here 'tasksManagement'?

File: src/pages/TaskDetailsPage.js [hunk 2]
```diff
@@ -33,13 +33,15 @@ function TaskDetailsPage({
   const history = useHistory();
   const { formatMessage } = useTranslations('tasksManagement', modulesManager);
   const [editedTask, setEditedTask] = useState({});
+  const [hasSaved, setHasSaved] = useState(false);
   const back = () => history.goBack();
 
   useEffect(() => {
     if (taskUuid) {
       fetchTask(modulesManager, [`id: "${taskUuid}"`]);
+      setHasSaved(false);
```
Comment (sniedzielski) [inline src/pages/TaskDetailsPage.js]: Maybe isSaved could be better when it comes to the naming

File: src/components/TaskSearcher.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  const fetchingTasks = useSelector((state) => state?.tasksManagement?.fetchingTasks);
+  const fetchedTasks = useSelector((state) => state?.tasksManagement?.fetchedTasks);
+  const errorTasks = useSelector((state) => state?.tasksManagement?.errorTasks);
+  const tasks = useSelector((state) => state?.tasksManagement?.tasks);
+  const tasksPageInfo = useSelector((state) => state?.tasksManagement?.tasksPageInfo);
+
+  const openTask = (task, newTab = false) => historyPush(
+    modulesManager,
+    history,
+    'tasksManagement.route.task',
```
Comment (sniedzielski) [inline src/components/TaskSearcher.js]: We can create const for that string value with that argument

**Review by sniedzielski — COMMENTED** [review]
Good job. I left some comments with some questions and some minor improvement of code. @jdolkowski

File: src/components/TaskPreviewTable.js (hunk 1 above)
Comment (jdolkowski) [reply src/components/TaskPreviewTable.js]: you are right, changed

File: src/components/TaskSearcher.js (hunk 3 above)
Comment (jdolkowski) [reply src/components/TaskSearcher.js]: done

File: src/pages/TaskDetailsPage.js (hunk 2 above)
Comment (jdolkowski) [reply src/pages/TaskDetailsPage.js]: done

**Review by sniedzielski — APPROVED** [review]
LGTM

**Pull request merged**